src/util/language.cpp

```diff
@@ -125,3 +129,74 @@ bool languaget::type_to_name(
   name=type.pretty();
   return false;
 }
+
+/*******************************************************************\
+
+Function: generate_start_function
+
+  Inputs:
+          entry_function_symbol - The symbol for the function that should
+                                  be used as the entry point
+          symbol_table - The symbol table for the program. The new _start
+                         function symbol will be added to this table
+
+ Outputs: Returns false if the _start method was generated correctly
+
+ Purpose: Generate a _start function for a specific function. Should
+          be overriden in derived languagets
+
+\*******************************************************************/
+bool languaget::generate_start_function(
+  const symbolt &entry_function_symbol,
+  symbol_tablet &symbol_table)
+{
+  // Implement in derived languagets
+  assert(0);
+  return true;
+}
```

Comment:
Maybe this is a lazy cop-out but I didn't make it abstract as don't know how to provide implementations for some of the languages. I could make it abstract and provide a stub method that asserts in the derived?